application-portal/src/app/services/choices/choices.service.ts

```diff
@@ -1,9 +1,284 @@
 import { Injectable } from '@angular/core';
+import { User } from '../../models/user/user'
+import { Program } from '../../models/program/program'
+import { Location } from '../../models/location/location'
+import { Period } from '../../models/period/period'
+import { Session } from '../../models/session/session'
+import { Choices } from '../../models/choices/choices';
 
 @Injectable({
   providedIn: 'root'
 })
 export class ChoicesService {
+  // these store current data
+  user: User;
+  program: Program;
+  location: Location;
+  period: Period;
+  session: Session;
+  choices: Choices;
 
-  constructor() { }
-}
+  // arrays that hold info from backend
+  programs: Array<Program>;
```

Comment:
you might consider naming these a bit more explicitly so you don't get confused, i.e. `allPrograms`, `allLocations`, etc.